src/wait-for.js

```diff
@@ -134,7 +137,7 @@ function waitFor(
     }
 
     function checkCallback() {
-      if (promiseStatus === 'pending') return
+      if (finished || promiseStatus === 'pending') return
```

Comment:
Can we cancel whatever is calling `checkCallback` instead?

Reply:
I would love to, but I'm afraid it might introduce breaking changes.

We'll need to move the `if(finished)` before `checkCallback` here: https://github.com/testing-library/dom-testing-library/pull/1271/files#diff-68584107d5f50eb95702a47b23bb6e0f6c2d78f9bab7456a1a4150c79229db78R84-R92

But the following comment is daunting: https://github.com/testing-library/dom-testing-library/pull/1271/files#diff-68584107d5f50eb95702a47b23bb6e0f6c2d78f9bab7456a1a4150c79229db78R84-R87

Reply:
Actually https://github.com/testing-library/dom-testing-library/pull/1271/files#diff-68584107d5f50eb95702a47b23bb6e0f6c2d78f9bab7456a1a4150c79229db78R84-R92 is un-related, so I should move the `if(finished)` before `checkCallback` 👍

Reply:
Amended in https://github.com/testing-library/dom-testing-library/pull/1271/commits/6b41483386804687e145119bc1966a2da711eb01